src/main/java/org/reactome/qa/doisuggester/ReactionlikeEvent.java

```diff
@@ -0,0 +1,354 @@
+package org.reactome.qa.doisuggester;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.InvalidAttributeException;
+
+/**
+ * Represents a Reactionlike Event, in the context of DOI suggestion activities.
+ * @author sshorser
+ *
+ */
+public class ReactionlikeEvent
+{
+	// The underlying RLE, in the current database.
+	private GKInstance rle;
+	// Corresponding RLE from the previous Release's database.
+	private ReactionlikeEvent prevRle = null;
+
+	private List<GKInstance> authoredInstancedEdits;
+	private List<GKInstance> reviewedInstancedEdits;
+	private List<GKInstance> revisedInstancedEdits;
+
+	private Set<GKInstance> ancestors = new HashSet<>();
+
+	/**
+	 * Constructor to create a DOISuggester-contexted Reaction-like Event object.
+	 * @param instance - The underlying instance from the database. This is a ReactionlikeEvent object.
+	 * @param prevAdaptor - A database adaptor that should point to the *previous* version of the database. This will be used to look at
+	 * previous versions of the <code>instance</code>
+	 * @throws Exception
+	 */
+	public ReactionlikeEvent(GKInstance instance, MySQLAdaptor prevAdaptor) throws Exception
+	{
+		this.rle = instance;
+		GKInstance prevInstance = prevAdaptor.fetchInstance(this.rle.getDBID());
+		if (prevInstance != null)
+		{
+			ReactionlikeEvent prevDOISuggesterRLE = new ReactionlikeEvent(prevInstance);
+			this.prevRle = prevDOISuggesterRLE;
+		}
+	}
+
+	/**
+	 * Returns true if the underlying RLE instance is new - there is no corresponding instance in the previous Releases's database.
+	 * @return
```

Comment:
For JavaDoc completeness, could the @return tag have a description (i.e. "returns <code>true</code> if there is no corresponding RLE in the previous release database; <code>false</code> otherwise")?

Reply:
That seems redundant.

Reply:
Also, if you don't escape the "@" in `@return`, it refers to an actual github user.

Reply:
Okay, thanks - I'll escape the the "@" from now on.  

I know it's redundant, but I an empty JavaDoc tag does not have a purpose and doesn't provide information.  If the return tag is present (which it should be for a public method), it should describe what is returned.

Maybe the description can be about the method as a whole and the circumstances of true/false can be written in the `@return` tag.  For methods like this, it's hard to avoid some redundancy but the tag should have a description.

For example:

```
/**
* Returns whether the RLE is new compared to a previous release database
* @return <code>true</code> if there is no corresponding RLE in the previous release database;<code>false</code> otherwise
* /
```

Reply:
Your example is just saying the same thing in two slightly different ways. I don't think it adds anything in this case.

I could remove the `@return` so there is no empty JavaDoc tag. ;)

Reply:
Yes, you're right.  In this case it's hard to avoid redundancy.  I try to make methods like this have a general description and then include details like "no RLE with the same dbId" in the return tag (or a specific description and a general return tag).

Removing the `@return` tag is fine, but I do think it makes sense to be there for public methods even if there is redundancy.  If you'd rather remove it, that's fine.

Reply:
Also, a non-void method will produce an error when compiling JavaDoc if no return tag exists, so I think it does make sense to have it there despite the redundancy.

Reply:
In that case, it seems that removing the `@return` tag is not a good idea, in case someone tries to generate JavaDoc files. :|